metatensor-torch/src/atomistic/model.cpp

```diff
@@ -1080,7 +1080,7 @@ bool metatensor_torch::valid_quantity(const std::string& quantity) {
             valid_quantities.emplace_back(it.first);
         }
 
-        TORCH_WARN(
+        TORCH_WARN_ONCE(
```

Comment:
You should use the same trick as here: https://github.com/metatensor/metatensor/blob/eeeafcd298a9caee9ad397fce47594f042816cdc/metatensor-torch/src/atomistic/system.cpp#L864-L870, since otherwise torch warns once per source location, not once per message.